src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -533,21 +531,17 @@ else if (ast.getType() == TokenTypes.SWITCH_RULE) {
     }
 
     /**
-     * Returns the last child token that makes a specified type and contains containType in
-     * its branch.
+     * Returns the last token of type {@link TokenTypes#CASE_GROUP} which contains
+     * {@link TokenTypes#SLIST}.
      *
-     * @param ast token to be tested
-     * @param childType the token type to match
-     * @param containType the token type which has to be present in the branch
+     * @param literalSwitchAst ast node of type {@link TokenTypes#LITERAL_SWITCH}
      * @return the matching token, or null if no match
      */
-    private static DetailAST findLastChildWhichContainsSpecifiedToken(DetailAST ast, int childType,
-                                                              int containType) {
+    private static DetailAST findLastCaseGroupWhichContainsSlist(DetailAST literalSwitchAst) {
         DetailAST returnValue = null;
-        for (DetailAST astIterator = ast.getFirstChild(); astIterator != null;
-                astIterator = astIterator.getNextSibling()) {
-            if (astIterator.getType() == childType
-                    && astIterator.findFirstToken(containType) != null) {
+        for (DetailAST astIterator = literalSwitchAst.getFirstChild(); astIterator != null;
+             astIterator = astIterator.getNextSibling()) {
+            if (astIterator.findFirstToken(TokenTypes.SLIST) != null) {
```

Comment:
as an optimization we could start with the last child, iterate to a previous sibling and stop when SLIST is found.
Not related to pitest coverage though, we can do this later.